### ISSUE TITLE
`range_start` and `select` classes do not work together when styling with tailwind css

### ISSUE THREAD

**Issue by berkerdemirer** [gpbl/react-day-picker#2309]
## Code 

```tsx
'use client';

import { DayPicker, type DayPickerProps } from 'react-day-picker';
import { cn } from '@/components/utils';
import { buttonVariants } from '@/components/primitives/button';
import { Icons } from '@/components/icons';

export type CalendarProps = DayPickerProps;

function Calendar({
  className,
  classNames,
  showOutsideDays = true,
  ...props
}: CalendarProps) {
  return (
    <DayPicker
      showOutsideDays={showOutsideDays}
      className={cn('p-4', className)}
      classNames={{
        month_grid: 'border-collapse space-y-1',
        months: 'inline-flex gap-4 w-full justify-center',
        month: 'space-y-4',
        day_button: `h-12 w-12 hover:border hover:border-primary hover:rounded-full disabled:hover:border-none`,
        range_start: 'bg-black rounded-none',
        selected: 'bg-primary text-primary-foreground rounded-full',
        range_middle: 'aria-selected:bg-primary/20 aria-selected:rounded-none',
        button_next: cn(
          buttonVariants({
            variant: 'ghost',
            className: 'h-10 w-10 bg-transparent p-0 opacity-100',
          })
        ),
        day: `h-12 w-12`,
        weekday: 'rounded-md w-12 h-12 font-[500] text-muted-foreground',
        button_previous: cn(
          buttonVariants({
            variant: 'ghost',
            className: 'h-10 w-10 bg-transparent p-0 opacity-100',
          })
        ),
        hidden: 'invisible',
        outside:
          'day-outside text-muted-foreground opacity-50 aria-selected:bg-primary/50 aria-selected:text-muted-foreground aria-selected:opacity-30',
        today: 'text-primary',
        disabled: 'text-muted-foreground opacity-50',
        caption_label: 'text-md font-bold',
        nav: 'flex items-center justify-between absolute w-full z-10 px-6',
        month_caption: 'flex justify-center items-center pt-2',
        ...classNames,
      }}
      components={{
        Chevron: (props) => {
          if (props.orientation === 'left') {
            return <Icons.chevronLeft {...props} />;
          }
          return <Icons.chevronRight {...props} />;
        },
      }}
      {...props}
    />
  );
}
Calendar.displayName = 'Calendar';

export { Calendar };
```

## Expected Behavior

During date range selection, I would expect `range_start` background not to replace `selected` background. I should have rounded circle for the `select` state and a square background for the `range_start` so that I can seamlessly connect rounded button to mid section like in the example provided in the docs.

## Actual Behavior

`range_start` class overrides the `select` background, therefore I cannot have the same behaviour as in the docs. 

## Screenshots

<img width="779" alt="image" src="https://github.com/user-attachments/assets/6d99b142-4df6-43c3-a91d-7ddd32cab857">

**Comment by gpbl** [comment]
Relevant discussion: https://github.com/gpbl/react-day-picker/discussions/2308

TLDR; there's a fix planned for this.

**Comment by gpbl** [comment]
As per https://github.com/gpbl/react-day-picker/discussions/2308#discussioncomment-10164633... moving back to the discussion. I don't think there're changes we should apply to DayPicker for this issue.

**Comment by vitorschelb** [comment]
After Hours trying, that the only way it worked for me:
modifiersStyles={{
range_start: {
backgroundColor: '#3182CE',
color: 'white',
borderRadius: '50%',
boxShadow: '10px 0px 0px #354A53',
position: 'relative',
zIndex: 2
},
range_end: {
backgroundColor: '#3182CE',
color: 'white',
borderRadius: '50%',
boxShadow: '-10px 0px 0px #354A53',
position: 'relative',
zIndex: 2
},
range_middle: {
backgroundColor: '#354A53',
boxShadow: '-10px 0px 0px #354A53, 10px 0px 0px #354A53',
position: 'relative',
zIndex: 1
}
}}
